llvm/lib/Transforms/Instrumentation/MemorySanitizer.cpp

```diff
@@ -4749,6 +4749,13 @@ struct MemorySanitizerVisitor : public InstVisitor<MemorySanitizerVisitor> {
       setOrigin(&I, getCleanOrigin());
       break;
 
+    // Saturating extract narrow
+    case Intrinsic::aarch64_neon_sqxtn:
+    case Intrinsic::aarch64_neon_sqxtun:
+    case Intrinsic::aarch64_neon_uqxtn:
+      handleShadowOr(I);
```

Comment:
Isn't the output only half as long as the original? Don't we need to account for that?

Also, it is confusing that you use `handleShadowOr` on a single argument instruction. At least leave a comment that explains.

Reply:
> Isn't the output only half as long as the original? Don't we need to account for that?

That's why I use handleShadowOr - it will perform the cast.

From the test output:
```
; CHECK-NEXT:    [[_MSPROP1:%.*]] = or <8 x i16> [[TMP1]], zeroinitializer
; CHECK-NEXT:    [[TMP5:%.*]] = trunc <8 x i16> [[_MSPROP1]] to <8 x i8>
```

> Also, it is confusing that you use handleShadowOr on a single argument instruction. At least leave a comment that explains.

I've added a comment.